src/services/EcoBridge/Arbitrum/ArbitrumBridge.tsx

```diff
@@ -0,0 +1,860 @@
+            blockNumber: undefined
+          })
+        )
+
+        this.store.dispatch(
+          this.actions.updateTxPartnerHash({
+            chainId: this.l2ChainId,
+            txHash: retryableTicketHash,
+            partnerTxHash: txn.txHash,
+            partnerChainId: this.l1ChainId
+          })
+        )
+      }
+    })
+  }
+
+  // Pending Withdrawals listener
+  private getOutgoingMessageState = async (tx: BridgeTxn) => {
+    const retVal: Partial<Pick<BridgeTxn, 'batchIndex' | 'batchNumber'>> &
+      Pick<BridgeTxn, 'txHash' | 'outgoingMessageState'> = {
+      batchNumber: tx.batchNumber,
+      batchIndex: tx.batchIndex,
+      outgoingMessageState: undefined,
+      txHash: tx.txHash
+    }
+
+    if (!tx.receipt) {
+      return retVal
+    }
+    if (!retVal.batchNumber || !retVal.batchIndex) {
+      const l2ToL2EventData = await this.bridge.getWithdrawalsInL2Transaction(tx?.receipt)
+      if (l2ToL2EventData.length === 1) {
+        const { batchNumber, indexInBatch } = l2ToL2EventData[0]
+        const outgoingMessageState = await this.bridge.getOutGoingMessageState(batchNumber, indexInBatch)
+        retVal.batchIndex = indexInBatch.toHexString()
+        retVal.batchNumber = batchNumber.toHexString()
+        retVal.outgoingMessageState = outgoingMessageState
+      }
+    } else {
+      const retValbatchNr = BigNumber.from(retVal.batchNumber)
+      const retValbatchIndex = BigNumber.from(retVal.batchIndex)
+      const outgoingMessageState = await this.bridge.getOutGoingMessageState(retValbatchNr, retValbatchIndex)
+      retVal.outgoingMessageState = outgoingMessageState
+    }
+
+    return retVal
+  }
+
+  private updatePendingWithdrawals = async () => {
+    if (this._initialPendingWithdrawalsChecked) return
+
+    this.store.dispatch(ecoBridgeUIActions.setBridgeLoadingWithdrawals(true))
+
+    const pendingWithdrawals = this.selectors.selectPendingWithdrawals(this.store.getState(), this._account)
+
+    const promises = pendingWithdrawals.map(this.getOutgoingMessageState)
+
+    const withdrawalsInfo = await Promise.all(promises)
+    withdrawalsInfo.forEach(withdrawalInfo => {
+      if (!this.l2ChainId) return
+      const { outgoingMessageState, batchNumber, batchIndex, txHash } = withdrawalInfo
+
+      if (outgoingMessageState !== undefined) {
+        this.store.dispatch(
+          this.actions.updateTxWithdrawal({
+            chainId: this.l2ChainId,
+            outgoingMessageState,
+            txHash,
+            batchIndex,
+            batchNumber
+          })
+        )
+      }
+    })
+
+    this.store.dispatch(ecoBridgeUIActions.setBridgeLoadingWithdrawals(false))
+    this._initialPendingWithdrawalsChecked = true
+  }
+
+  // Handlers
+  private depositEth = async (value: string) => {
+    if (!this._account) return
+
+    this.store.dispatch(ecoBridgeUIActions.setBridgeModalStatus({ status: BridgeModalStatus.PENDING }))
+    const weiValue = utils.parseEther(value)
```

Comment:
Let's import `parseEther` without `utilts` since `parseUnits` is imported.

Reply:
done

Reply:
Resolved by #874